simplified-books-controller/src/main/java/org/nypl/simplified/books/controller/BookSyncTask.kt

```diff
@@ -64,12 +66,14 @@ class BookSyncTask(
     val providerAuth = provider.authentication
     if (providerAuth == AccountProviderAuthenticationDescription.Anonymous) {
       this.logger.debug("account does not support syncing")
+      this.removeExpiredBooks(account)
       return this.taskRecorder.finishSuccess(Unit)
     }
 
     val credentials = account.loginState.credentials
     if (credentials == null) {
       this.logger.debug("no credentials, aborting!")
+      this.removeExpiredBooks(account)
```

Comment:
This case and the one above at L69 are the no-auth accounts that do not support remote syncing.